easy_phi/tests/handlers_test.py

```diff
@@ -213,3 +216,31 @@ def test_mime_type(self):
             "ModuleUI handler returned wrong content type "
             "('application/javascript' expected)"
         )
+
+class WebSocketBaseTestCase(tornado.testing.AsyncHTTPTestCase):
+    @gen.coroutine
+    def ws_connect(self, path, compression_options=None):
+        ws = yield websocket_connect(
+            'ws://127.0.0.1:%d%s' % (self.get_http_port(), path),
+            compression_options=compression_options)
+        raise gen.Return(ws)
+
+    @gen.coroutine
+    def close(self, ws):
+        """Close a websocket connection and wait for the server side.
+        If we don't wait here, there are sometimes leak warnings in the
+        tests.
+        """
+        ws.close()
+
+class WebSocketTest(WebSocketBaseTestCase):
+    def get_app(self):
+        return app.application
+
+    @tornado.testing.gen_test
+    def test_websocket_gen(self):
+        ws = yield self.ws_connect('/websocket')
```

Comment:
it is not a real test. You specifically added handler in the main app just to test it, while rest remains untouched.
Try to trigger events in the app and listen for changes - this is our main usecase